packages/cli/README.md

```diff
@@ -1,66 +1,789 @@
+export const builder = (yargs) => {
+  return yargs.option('u', {
+    alias: 'url',
+    describe: 'the URL to make an HTTP request to'
+  })
+}
+export const handler = (argv) => {
+  console.log(argv.url)
+}
+```
+
+To get a good sense of the difference, compare [redwood-tools.js](https://github.com/redwoodjs/redwood/blob/d51ade08118c17459cebcdb496197ea52485364a/packages/cli/src/redwood-tools.js#L79-L167) to [dev.js](https://github.com/redwoodjs/redwood/blob/d51ade08118c17459cebcdb496197ea52485364a/packages/cli/src/commands/dev.js#L10-L77), or any other command.
+
+### Overview
+
+Contributing to `@redwoodjs/cli` usually means adding a command or modifying an existing one. We've organized this doc around adding a command since if you know how to do this you'll know how to modify one too.
+
+### Best Practices
+
+There's a few best practices we follow that you should be aware of:
+
+- **Use `description` instead of `desc` or `describe`:** While yargs accepts any of these, for consistency, use `description`
+- **descriptions shouldn't end in periods:** Again, just a stylistic choice&mdash;but stick to it!
+- **`builder` should be a function:** This enables the positional api (more on that [later](#builder))
+- **Update the docs and test:** By docs, we mean the content on redwoodjs.com. We know this can be a lot, so don't feel like you have to do it all yourself. We're more than happy to help&mdash;just ask us!
+
+If none of these make sense yet, don't worry! You'll see them come up in the next section, where we walk you through adding a command.
+
+### Adding a Command
+
+You can add a command by creating a file in [./src/commands](https://github.com/redwoodjs/redwood/tree/master/packages/cli/src/commands). Although it's not necessary, for consistency, the file should be named after the command that invokes it. For example, the build command, which is invoked with
 
 ```terminal
-yarn create redwood-app <project-dir>
+yarn rw build 
 ```
 
-OK, OK, so this isn't really part of the RedwoodJS CLI, per se. But we felt it belonged here anyway!
+lives in [./src/commands/build.js](https://github.com/redwoodjs/redwood/blob/master/packages/cli/src/commands/build.js).
+
+<!-- [todo] -->
+<!-- Add alias here? Not required tho -->
+To make a command using the advanced api, yargs requires that you export [four constants](https://github.com/yargs/yargs/blob/master/docs/advanced.md#providing-a-command-module):
 
-We create a new Redwood application by running the `yarn create redwood-app <project-dir>` command, where `<project-dir>` is the path of the to-be Redwood project.
+|Constant|Type|Description|
+|:-|:-|:-|
+|`command`|string|A yargs command definition. You specify the command's name and arguments here|
+|`description`|string|A description of the command; shown in the help message|
+|`builder`|function|Effectively "builds" the command's arguments/options|
+|`handler`|function|The function invoked by the command; does all the work|
```

Comment:
I think there's a markdown formatter available in vscode that'll make these easier to read.

Reply:
Thanks, just did that. Looks so much better. I 'll have to do that across the docs/from now on.

Turns out I had the extension already but wasn't using the keyboard shortcut to trigger the formatter.